Library/Homebrew/github_runner_matrix.rb

```diff
@@ -118,6 +120,15 @@ def create_runner(platform, arch, spec, macos_version = nil)
     runner.freeze
   end
 
+  NEWEST_HOMEBREW_CORE_MACOS_RUNNER = :sonoma
+  OLDEST_HOMEBREW_CORE_MACOS_RUNNER = :monterey
+  NEWEST_HOMEBREW_CORE_INTEL_MACOS_RUNNER = :sonoma
+
+  sig { params(macos_version: MacOSVersion).returns(T::Boolean) }
+  def runner_enabled?(macos_version)
+    macos_version <= NEWEST_HOMEBREW_CORE_MACOS_RUNNER && macos_version >= OLDEST_HOMEBREW_CORE_MACOS_RUNNER
+  end
+
   NEWEST_GITHUB_ACTIONS_INTEL_MACOS_RUNNER = :ventura
   OLDEST_GITHUB_ACTIONS_INTEL_MACOS_RUNNER = :big_sur
```

Comment:
```suggestion
  OLDEST_GITHUB_ACTIONS_INTEL_MACOS_RUNNER = :monterey
```

GitHub have dropped support for `macos-11` now.

https://github.blog/changelog/2024-05-20-actions-upcoming-changes-to-github-hosted-macos-runners/